### PR TITLE
Refactor integration tests

### PR DESCRIPTION
**Pull request by odeac**
## Overview
While writing the readme.md some aspects were more difficult to explain than to actually change the code and make it simpler. Hence this refactoring PR.

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
https://rchain.atlassian.net/browse/RHOL-770

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You assigned one person to review this PR

### If you are not a member of the core development team, please confirm:
- [ ] You signed the commit. Merging requires a signature. Please see the [developer wiki](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain) for instructions.
- [ ] Your GitHub account is also an account with [Travis CI](https://travis-ci.org). Unit tests will not run on your PR unless you have an account with Travis. Merge requires passed unit tests.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: integration-testing/src/tools/util.py [hunk 1]
```diff
@@ -24,3 +28,97 @@ def make_tempfile(prefix, content):
...
+
+    def make_timeout(value, base, peer_factor=10): return value if value > 0 else base + peer_count * peer_factor
+
+    config = Config( peer_count = peer_count,
+                     node_startup_timeout = make_timeout(start_timeout, 30, 10),
+                     network_converge_timeout = make_timeout(converge_timeout, 200, 10),
+                     receive_timeout = make_timeout(receive_timeout, 10, 10),
+                     rnode_timeout = rnode_timeout,
+                     blocks = blocks
+                     )
```
Comment (adaszko) [inline integration-testing/src/tools/util.py]: Non-PEP8 formatting.

File: integration-testing/src/tools/util.py [hunk 2]
```diff
@@ -24,3 +28,97 @@ def make_tempfile(prefix, content):
...
+    f=open(resources.file_path('pregenerated-validator-private-public-key-pairs.txt'))
+    lines=f.readlines()
+    random.shuffle(lines)
+    validator_keys = [ KeyPair(*line.split())
+                       for line in lines[0:config.peer_count+1]]
+
+    logging.info(f"Using validator keys: {validator_keys}")
+
+    with bonds_file(validator_keys) as f:
+        yield ValidatorsData(f, validator_keys[0], validator_keys[1:])
```
Comment (adaszko) [inline integration-testing/src/tools/util.py]: Passing by keyword arguments here would be nice to have for clarity.

File: integration-testing/src/tools/util.py [hunk 3]
```diff
@@ -24,3 +28,97 @@ def make_tempfile(prefix, content):
...
+    # WARN  coop.rchain.casper.Validate$ - CASPER: Ignoring block 2cb8fcc56e... because block creator 3641880481... has 0 weight
+    f=open(resources.file_path('pregenerated-validator-private-public-key-pairs.txt'))
+    lines=f.readlines()
+    random.shuffle(lines)
+    validator_keys = [ KeyPair(*line.split())
+                       for line in lines[0:config.peer_count+1]]
+
+    logging.info(f"Using validator keys: {validator_keys}")
+
+    with bonds_file(validator_keys) as f:
```
[the remark this answers is not in the document]
Comment (odeac) [reply integration-testing/src/tools/util.py]: done

File: integration-testing/src/tools/util.py (hunk 2 above)
Comment (odeac) [reply integration-testing/src/tools/util.py]: done

File: integration-testing/src/tools/util.py [hunk 4]
```diff
@@ -24,3 +29,95 @@ def make_tempfile(prefix, content):
...
+
+@contextmanager
+def bonds_file(validator_keys):
+    (fd, file) = tempfile.mkstemp(prefix="rchain-bonds-file-", suffix=".txt", dir="/tmp")
+    logging.info(f"Using bonds file: `{bonds_file}`")
+
+    with os.fdopen(fd, "w") as f:
+        for pair in validator_keys:
+            bond = random.randint(1, 100)
+            f.write(f"{pair.public_key} {bond}\n")
```
Comment (adaszko) [inline integration-testing/src/tools/util.py]: I think if one of those writes fail, the temporary file won't be deleted.

File: integration-testing/src/tools/util.py [hunk 5]
```diff
@@ -24,3 +29,95 @@ def make_tempfile(prefix, content):
...
+
+    random.shuffle(lines)
+
+    validator_keys = [KeyPair(*line.split())
+                      for line in lines[0:config.peer_count+1]]
+
+    logging.info(f"Using validator keys: {validator_keys}")
+
+    with bonds_file(validator_keys) as f:
+        yield ValidatorsData(bonds_file=f, boostrap_keys=validator_keys[0], peers_keys=validator_keys[1:])
```
Comment (adaszko) [inline integration-testing/src/tools/util.py]: 👍

File: integration-testing/src/tools/util.py (hunk 4 above)
Comment (odeac) [reply integration-testing/src/tools/util.py]: good point

File: integration-testing/src/tools/util.py (hunk 1 above)
Comment (odeac) [reply integration-testing/src/tools/util.py]: done

File: integration-testing/src/tools/util.py [hunk 6]
```diff
@@ -24,3 +28,97 @@ def make_tempfile(prefix, content):
...
+    finally:
+        os.unlink(file)
+        logging.info(f"Bonds file `{bonds_file}` deleted")
+
+
+@contextmanager
+def validators_data(config):
+    # Using pre-generated validator key pairs by rnode. We do this because warning below  with python generated keys
+    # WARN  coop.rchain.casper.Validate$ - CASPER: Ignoring block 2cb8fcc56e... because block creator 3641880481... has 0 weight
+    f=open(resources.file_path('pregenerated-validator-private-public-key-pairs.txt'))
```
[the remark this answers is not in the document]
Comment (odeac) [reply integration-testing/src/tools/util.py]: fixed

File: integration-testing/test/conftest.py [hunk 7]
```diff
@@ -125,4 +23,17 @@ def pytest_addoption(parser):
...
+
+
+
+@pytest.fixture(scope="session")
+def system(request):
+    cfg = parse_config(request)
+
+    with docker() as docker_client, \
+         validators_data(cfg) as vd:
+            try:
```
[the remark this answers is not in the document]
Comment (odeac) [reply integration-testing/test/conftest.py]: fixed

**Pull request merged**